### ISSUE TITLE
Is it even possible to add a command to flash the dump?

### ISSUE THREAD

**Issue by MorfiMorfi** [dualshock-tools/ds4-tools#37]
it turned out that I was doing the calibration with the battery disconnected and it looks like something broke. The gamepad works only by wire, Bluetooth refuses to work and the gamepad does not even light up without connecting the wire. Although even by wire it only works with a PC, on ps4 the gamepad only works in the service menu.

**Comment by dualshock-tools** [comment]
@MorfiMorfi this whole project started to fix a DS4 i had with exactly that issue. 
At the moment there is no way to reflash a dump, until we figure out how to do it.

The DS4 has a major flaw: there is a checksum of the configuration which should always match. If the DS4 is disconnected while updating the configuration (e.g. while pairing with the PS4), at the next boot it suicides itself by clearing completely the config and, unless you have internal tools by Sony, it becomes unusable.

That is the damage described in all the parts of this blog https://blog.the.al 
There are some attempts to revive it, but nothing for bluetooth or battery charging

**Comment by qwertychouskie** [comment]
> @MorfiMorfi this whole project started to fix a DS4 i had with exactly that issue. At the moment there is no way to reflash a dump, until we figure out how to do it.
> 
> The DS4 has a major flaw: there is a checksum of the configuration which should always match. If the DS4 is disconnected while updating the configuration (e.g. while pairing with the PS4), at the next boot it suicides itself by clearing completely the config and, unless you have internal tools by Sony, it becomes unusable.
> 
> That is the damage described in all the parts of this blog [blog.the.al](https://blog.the.al) There are some attempts to revive it, but nothing for bluetooth or battery charging

IMHO there should probably be a warning that says something about the fact that you should always calibrate with a working and charged battery plugged in to reduce the risk of potential bricks.  Also, it would be nice if, before the calibration, the software backed up the values (e.g. gyro cal data, Bluetooth mac address, etc) that are prone to wipe upon failure.

**Comment by dualshock-tools** [comment]
@qwertychouskie you are right. yesterday I added a warning to the web-ui and will add soon the same to the scripts. unfortunately, the backed up values need more reverse engineering to be read correctly. 

@MorfiMorfi when I will manage to fix my DS4 (stuck in a serial mode now), I will continue to try to restore the values, so that this can become a feature of the web-ui. it will probably take a year or more, but I'm hopeful that at some point your DS4 will be restored

**Comment by carpikes** [comment]
@MorfiMorfi send me an email at ds4@the.al, i may be able to revive your controller

**Comment by rodolfostanic** [comment]
Hopeful to be able to restore my DS4 to its former state in the future. Unfortunately, after calibration, I'm having all kinds of issues with bluetooth. 😢
